### ISSUE TITLE
As an Open Data Hub Mobility user I would like to have the possibility, through a proper API call, to get also the historical metadata records associated to a certain station

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/it.bz.opendatahub.api.mobility-ninja#30]

**Comment by clezag** [comment]
@rcavaliere A first go at the implementation is now [online in testing](https://mobility.api.opendatahub.testingmachine.eu/v2/tree%2Cnode/%2A/metadata/2023-05-01/2023-05-10?limit=200&offset=0&shownull=false&distinct=true&timezone=UTC):

You can get to the historical metadata via a level 5 call (e.g. flat/StationType/DataType/from/to), specifying "metadata" as the datatype:

e.g. flat/EchargingStation/metadata/2023-01-01/2023-05-15

The fields are then returned with prefix "mh" for metadata history:
- mhtransactiontime: creation date
- mhdata: the metadata json

Note that the current station and parent metadata are still present (we could change this).

**Comment by rcavaliere** [comment]
@clezag in my opinion this OK!

**Comment by rcavaliere** [comment]
@clezag one question: do you include also in the set of fields in metadata also the basic fields in station, i.e. what we decided here: https://github.com/noi-techpark/bdp-core/issues/269? In general I would put all relevant fields in metadata as well, not only scode but also sname, scoordinate.

**Comment by rcavaliere** [comment]
@clezag no, I would do this for all type of stations, so that we always have in the metadata also the basic fields in the station table

**Comment by clezag** [comment]
Released the new API implementation in production

**Comment by rcavaliere** [comment]
@clezag didn't you integrate in the set of metadata all the station fields, or?

**Comment by clezag** [comment]
@rcavaliere no, only the request API for metadata history is currently live, the other part has to be done on the writer (inbound) side
